### PR TITLE
[BXMSPROD-1843] RHBOP UMB Message Job

### PR DESCRIPTION
**Pull request by lampajr**
**Thank you for submitting this pull request**

**JIRA**: 
- https://issues.redhat.com/browse/BXMSPROD-1843

**referenced Pull Requests**: 
- https://github.com/kiegroup/jenkins-pipeline-shared-libraries/pull/220

Fixes:
- Add new job for `RHBOP` product aiming to compute and send the UMB message to QE.
- UMB trigger jobs refactoring in order to use newly exposed `pnc` lib (see https://github.com/kiegroup/jenkins-pipeline-shared-libraries/pull/220)

> Note: once https://github.com/kiegroup/jenkins-pipeline-shared-libraries/pull/220 will be merged, we can merge this too.

<pre>
How to retest a PR or trigger a specific build:

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
</pre>

### CODE REVIEW

File: job-dsls/src/main/resources/job-scripts/prod_rhbop_trigger_umb_message.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import groovy.json.JsonSlurper
+import java.net.URLEncoder
+
+// Projects to find the productized version in a milestone
+def projects = ["kiegroup/drools", "kiegroup/optaplanner"]
+// ID of Red Hat build of OptaPlanner product in PNC.
+def productId = "${env.PRODUCT_ID.trim() ?: 'TODO'}"
```
Comment (lampajr) [inline job-dsls/src/main/resources/job-scripts/prod_rhbop_trigger_umb_message.jenkinsfile:7]: We will only know the RHBOP id when we first run a build on PNC

File: job-dsls/src/main/resources/job-scripts/prod_rhbop_trigger_umb_message.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    return projectsAndVersions.isEmpty() ? false : !projectsAndVersions.find{it.value == null} as Boolean
+}
+
+// generate UMB message body
+def getMessageBody(milestone, projectsAndVersions) {
+    // TODO: retrieve built projects from projectsAndVersions keys
+    return """
+{
+    "built_projects": ["drools","optaplanner","optaplanner-quickstarts","optaweb-vehicle-routing","jboss-integration/rhbop-optaplanner"],
+    "maven_repository_file_url": "${env.STAGING_SERVER_URL}/rhbop/RHBOP-${milestone}/rhbop-${milestone}-maven-repository.zip",
```
[the remark this answers is not in the document]
Comment (mareknovotny) [reply job-dsls/src/main/resources/job-scripts/prod_rhbop_trigger_umb_message.jenkinsfile:80]: that is a good note.

[the remark this answers is not in the document]
Comment (lampajr) [reply job-dsls/src/main/resources/job-scripts/prod_rhbop_trigger_umb_message.jenkinsfile:80]: I guess you are right! Adding it.

**Review by rgdoliveira — APPROVED** [review]
LGTM

**Pull request merged**